server/middleware/stage.js

```diff
@@ -30,48 +27,42 @@ async function stageApiResponseMiddleware(req, res, next) {
 }
 
 async function allocateCase(req, res, next) {
-    const logger = getLogger(req.request);
     const { caseId, stageId } = req.params;
     const { user } = req;
     try {
         await caseworkService.put(`/case/${caseId}/stage/${stageId}/user`, {
             userUUID: user.uuid,
         }, { headers: User.createHeaders(user) });
     } catch (error) {
-        logger.error(error);
-    } finally {
-        next();
+        return next(error);
     }
+    next();
 }
 
 async function allocateCaseToTeamMember(req, res, next) {
-    const logger = getLogger(req.request);
     const { caseId, stageId } = req.params;
     try {
         await caseworkService.put(`/case/${caseId}/stage/${stageId}/user`, {
             userUUID: req.body['user-id'],
         }, { headers: User.createHeaders(req.user) });
     } catch (error) {
-        logger.error(error);
-    } finally {
-        next();
+        return next(error);
     }
+    next();
+
 }
 
 async function moveByDirection(req, res, next) {
-    const logger = getLogger(req.request);
     const { caseId, stageId, flowDirection } = req.params;
     const { user } = req;
     try {
         await workflowService.post(`/case/${caseId}/stage/${stageId}/direction/${flowDirection}`, {
             userUUID: user.uuid,
         }, { headers: User.createHeaders(user) });
     } catch (error) {
-        logger.error(error);
-        next(error);
-    } finally {
-        next();
+        return next(error);
     }
+    next();
```

Comment:
Probably best to not return a function sometimes and undefined other times?
```suggestion
    return next();
```

Reply:
(the same for the other functions)

Reply:
Doesn't actually have an impact but I've done it to make it clear to anyone else in the future...